### PR TITLE
DM-38955: Fix composite functor error log and fix bad EBV units.

### PR DESCRIPTION
**Pull request by erykoff**
This also adds associated tests.

### CODE REVIEW

File: python/lsst/pipe/tasks/functors.py [hunk 1]
```diff
@@ -479,7 +480,13 @@ def __call__(self, data, **kwargs):
                     )
                     valDict[k] = f._func(subdf)
                 except Exception as e:
-                    self.log.error("Exception in %s call: %s: %s", self.name, type(e).__name__, e)
+                    self.log.error(
```
Comment (timj) [inline python/lsst/pipe/tasks/functors.py:483]: Are you sure you don't want to use `self.log.exception` here and automatically report the stack trace? If someone sees an exception like this inside the functor code will they be able to work out where their code triggered it?

Comment (erykoff) [reply python/lsst/pipe/tasks/functors.py:483]: I am not sure about that, I just changed this to work better than it was working.  How does one use `self.log.exception`?

Comment (timj) [reply python/lsst/pipe/tasks/functors.py]: Same as `self.log.error` except it automatically includes the exception information.

**Review by timj — APPROVED** [review]
I can see the name is now defined and the logging is better.

I have no idea about the functors themselves and the tests but they look fine.

**Pull request merged**